exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillStatsTable.java

```diff
@@ -134,6 +135,33 @@ public Double getRowCount() {
     return rowCount > 0 ? rowCount : null;
   }
 
+  /**
+   * Get the approximate number of distinct values of given column. If stats are not present for the
+   * given column, a null is returned.
+   *
+   * Note: returned data may not be accurate. Accuracy depends on whether the table data has changed
+   * after the stats are computed.
+   *
+   * @param col - column for which approximate count distinct is desired
+   * @return approximate count distinct of the column, if available. NULL otherwise.
+   */
+  public Double getNNRowCount(String col) {
+    // Stats might not have materialized because of errors.
+    if (!materialized) {
+      return null;
+    }
+    final String upperCol = col.toUpperCase();
+    Long nnRowCntCol = nnRowCount.get(upperCol);
+    if (nnRowCntCol == null) {
+      nnRowCntCol = nnRowCount.get(SchemaPath.getSimplePath(upperCol).toString());
```

Comment:
I think the SchemaPath method was use to correctly construct names for nested scalar columns. Maybe it would also work for simple columns but I did not try it. I can make this change later on.